modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/scim2/SCIM2UserTestCase.java

```diff
@@ -325,6 +336,33 @@ private void validateFilteredUser(String attributeName, String operator, String
         assertEquals(userId, this.userId);
     }
 
+    private void validateFilteredUserByEmailAddresses(String attributeName, String operator, String attributeValue)
+            throws IOException {
+
+        String userResourcePath = getPath() + "?filter=" + USER_SYSTEM_SCHEMA_ATTRIBUTE + ":" + attributeName + operator
+                + attributeValue;
+        HttpGet request = new HttpGet(userResourcePath);
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, "application/json");
+        request.addHeader(HttpHeaders.ACCEPT, "application/scim+json");
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 200, "User " +
+                "has not been retrieved successfully");
+
+        Object responseObj = JSONValue.parse(EntityUtils.toString(response.getEntity()));
+        EntityUtils.consume(response.getEntity());
+
+        JSONArray emailsArray = ((JSONArray)((JSONObject)((JSONObject) ((JSONArray) ((JSONObject) responseObj)
```

Comment:
is this much casting necessary?

Reply:
In this case a custom attribute (emailAddresses) is processed. It needs to be accessed via the custom schema hence the casting has to be done.
Improve the logic by breaking down the casting operations.
https://github.com/wso2/product-is/pull/23619/commits/2e4dd291689b3e9fd029f966b7095f03f49c5ff5